onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -53,12 +53,25 @@ def process_tableau_data(data, xform):
     with the column header fields for the same form.
     Handles Flattenning repeat data for tableau
     """
+
+    def get_xpath(key, nested_key):
+        val = nested_key.split('/')
+        nested_key_diff = val[len(key.split('/')):]
+        xpaths = key + f'[{index}]/' + '/'.join(nested_key_diff)
+        return xpaths
+
+    def unpack_data(key, value, data=None, xpaths=None):
```

Comment:
Please add a docstring for this function.... Finding it a bit hard to tell why `xpaths` is an argument... In what scenario would someone pass xpaths ?

_Is data supposed to be optional ? Seems like the function can't run without it ?_

Reply:
Ah...changed those two items. Thanks for the catch